python/src/plugins/headless/goat_plugins/crossmint_headless_checkout/service.py

```diff
@@ -0,0 +1,316 @@
+                            "data": to_hex(decoded[5]) if decoded[5] else "0x",
+                        }
+                    except Exception:
+                        # If RLP fails, try to handle as typed transaction by removing type byte
+                        if tx_bytes[0] <= 0x7f:
+                            # Remove the type byte and try RLP on the rest
+                            try:
+                                decoded = rlp.decode(tx_bytes[1:])
+                                # EIP-1559 format is similar but with different fields
+                                transaction_data = {
+                                    "to": to_hex(decoded[5]) if len(decoded) > 5 and decoded[5] else None,
+                                    "value": int.from_bytes(decoded[6], 'big') if len(decoded) > 6 and decoded[6] else 0,
+                                    "data": to_hex(decoded[7]) if len(decoded) > 7 and decoded[7] else "0x",
+                                }
+                            except Exception:
+                                raise Exception("Failed to decode transaction - unsupported format")
+                        else:
+                            raise Exception("Failed to decode transaction")
+                    
+                    if not transaction_data.get("to"):
+                        raise Exception("Transaction 'to' field is null or empty")
+                    
+                    logger.info(f"Parsed transaction: to={transaction_data['to']}, value={transaction_data['value']}, data={transaction_data['data'][:20]}...")
+                    
+                except Exception as parse_error:
+                    logger.error(f"Failed to parse transaction: {parse_error}")
+                    # Fallback: try to send raw transaction if parsing fails
+                    if hasattr(wallet_client, '_web3'):
+                        tx_hash = wallet_client._web3.eth.send_raw_transaction(serialized_transaction)
+                        receipt = wallet_client._web3.eth.wait_for_transaction_receipt(tx_hash)
+                        result = {
+                            "hash": receipt["transactionHash"].hex(),
+                            "status": "1" if receipt["status"] == 1 else "0",
+                        }
+                        return {"order": order, "txId": result["hash"]}
+                    else:
+                        raise Exception(f"Failed to parse transaction and no fallback available: {parse_error}")
+                
+                # Send the parsed transaction using the wallet client's send_transaction method
+                if hasattr(wallet_client, 'send_transaction'):
+                    import asyncio
+                    import inspect
+                    
+                    # Check if send_transaction is async
+                    send_tx_method = getattr(wallet_client, 'send_transaction')
+                    if inspect.iscoroutinefunction(send_tx_method):
```

Comment:
I dont think you need this, checkout how we send the transaction in the Uniswap python plugin, the swap tokens tool